### PR TITLE
Implement list functionality

### PR DESCRIPTION
**Pull request by adammw**
cc @uthark @SerenaTiede-Zen @grosser

### CODE REVIEW

**Review by Timothy-Dougherty — APPROVED** [review]

File: dynakube.go [hunk 1]
```diff
@@ -112,7 +116,36 @@ func (c *Client) Get(ctx context.Context, key client.ObjectKey, obj client.Objec
...
-	panic("implement me")
+	listOpts := client.ListOptions{}
+	listOpts.ApplyOptions(opts)
+	metav1ListOpts := listOpts.AsListOptions()
+	if metav1ListOpts == nil {
+		return fmt.Errorf("unable to convert %T to %T", listOpts, metav1ListOpts)
+	}
+
+	gvk, _ := apiutil.GVKForObject(list, c.scheme)
+	gvk.Kind = gvk.Kind[:len(gvk.Kind)-4] /*base library appends List*/
```
Comment (grosser) [inline dynakube.go:127]: this is wild ...

File: dynakube.go [hunk 2]
```diff
@@ -112,7 +116,36 @@ func (c *Client) Get(ctx context.Context, key client.ObjectKey, obj client.Objec
...
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+
+	// TODO: support non-namespaced resources
```
Comment (grosser) [inline dynakube.go:130]: can it panic if used on non-namespaced or would it fail ? ... maybe add a failing test ?

File: dynakube.go [hunk 3]
```diff
@@ -112,7 +116,36 @@ func (c *Client) Get(ctx context.Context, key client.ObjectKey, obj client.Objec
...
+	var action testing.Action
+	if listOpts.Namespace == "" {
```
Comment (grosser) [inline dynakube.go:132]:
```suggestion
	if listOpts.Namespace == "" { // asking for resources from all namespaces ?
```

**Review by grosser — APPROVED** [review]
looks about right 👍

File: dynakube.go (hunk 1 above)
Comment (adammw) [reply dynakube.go:127]: you'll never guess where I got it from: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/dynamic/fake/simple.go#L356

File: dynakube.go (hunk 2 above)
Comment (adammw) [reply dynakube.go:130]: problem is we can't tell unless we have the proper restmapper in place (which currently panics/is a TODO)

**Review by uthark — APPROVED** [review]

**Pull request merged**